src/categorical_algebra/FinCats.jl

```diff
@@ -104,17 +103,81 @@ is_discrete(C::FinCat) = isempty(hom_generators(C))
 """
 is_free(C::FinCat) = isempty(equations(C))
 
+"""Graph underlying a finitely presented category whose 
+object and
+hom generators are indexable, other
+than one explicitly generated by a graph.
+"""
+function graph(C::FinCat;maps::Bool=false,inv::Bool=false,named::Bool=false) 
+  g =  DiagramGraph() 
+  obgens = ob_generators(C)
+  homgens = hom_generators(C)
+  add_vertices!(g,length(obgens);vname=nameof.(obgens))
+  s(f) = only(indexin([dom(f)],obgens))
+  t(f) = only(indexin([codom(f)],obgens))
+  add_edges!(g,Int[s(f) for f in homgens],Int[t(f) for f in homgens];ename=nameof.(homgens)) 
+  named || begin g′ = Graph(); copy_parts!(g′,g,(:V,:E)); g = g′ end
+  maps || return g
+  inv ? (g, Dict(obgens[i] => i for i in 1:length(obgens)),
+  Dict(homgens[i] => i for i in 1:length(homgens))) :
+  (g, Dict(pairs(obgens)), Dict(pairs(homgens)))
+end
+#probably move into Graphs
```

Comment:
Agreed that this does not belong here.

Reply:
Alright, I moved it into BasicGraphs, non-exported, and imported it, DiagramGraph, and the four vertex/edge_name(d) methods as needed.